### ISSUE TITLE
Integrate the header files from the `include` directory into `src`

### ISSUE THREAD

**Issue by johnnovak** [dosbox-staging/dosbox-staging#2757]
Having separate `include` and `src` directories makes a lot of sense for libraries where the public header files from `include` and the object files are your published artifacts.

It makes very little sense for _applications_, though, where nothing gets published and your final deliverable is a binary (plus resource files and 3rd party DLLs, of course, but no header files).

See question and top comments here:
https://stackoverflow.com/questions/21881410/best-practice-src-include-or-include-folder-next-to-src-folder

The separate `include` dir of DOSBox serves no useful purpose; it's pretty arbitrary whether something goes into `src` or `include`. It's all over the place currently, and yeah, header-only stuff mostly lives in include, but that's no good reason for its existence.

As this separation only complicates and confuses things but solves nothing, it has to go. So I'm proposing to integrate all header files from `include` into `src`, then get rid of the `include` dir altogether.

Note I said _integrate_, not _dump_ 😄 Simply moving all header files into `src` and calling it a day won't cut it; every individual file must be moved to their most appropriate place.

**Comment by FeralChild64** [comment]
I have some doubts. Our _application_ in reality consists of several different _modules_, and it makes sense to have a separate directory for inter-module communication headers.
If you look into `src/hardware/input` - there are quite a few headers there, but they are used for communication inside mouse/keyboard emulation only. `include/keyboard.h` or `include/mouse.h`  are to be used by external modules. Sure, this separation is not 100% clean and it could be improved - but there is clearly an idea behind it.

**Comment by johnnovak** [comment]
Yeah, I'm agreeing with you on the importance of making a distinction between "module-internal public" and proper "module public" exports. But that further solidifies

**Comment by johnnovak** [comment]
(The cat closed the issue, not me 😹 ⌨️ )

**Comment by johnnovak** [comment]
Addressed in https://github.com/dosbox-staging/dosbox-staging/pull/4451
